hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java

```diff
@@ -188,6 +189,9 @@
      */
     Iterator<Record> iterator(long now, boolean backup);
 
+
+    MapKeyIteratorResult iterate(int tableIndex, int size);
+
```

Comment:
Hmm, it's not really doing an iteration. I mean like a standard  `iterator` that people are used to when they see the word `iterator`.
It's just fetching keys and returning the next table index to read from.
The returning of the nextTableIndex is like  side-effect to me so IMHO I would not concentrate on that while naming this method. Maybe just `fetchKeys` or sth along these lines?